packages/@aws-cdk/aws-dynamodb/lib/replica-provider.ts

```diff
@@ -27,10 +27,36 @@ export class ReplicaProvider extends NestedStack {
    */
   public static getOrCreate(scope: Construct, props: ReplicaProviderProps = {}) {
     const stack = Stack.of(scope);
-    const uid = '@aws-cdk/aws-dynamodb.ReplicaProvider';
+    const uid = this.getUid(stack);
     return stack.node.tryFindChild(uid) as ReplicaProvider ?? new ReplicaProvider(stack, uid, props);
   }
 
+  public static clearUids() {
+    this.getOrCreateCalls.clear();
+  }
```

Comment:
Hmm should this be exposed? I understand that it is needed for tests, but maybe in tests you could use another stackName to check results?

Reply:
It's not really exposed because the file `replica-provider.ts` is not exported from `index.ts`.